include/logger.h

```diff
@@ -58,12 +68,15 @@ template <const char *Fmt> class Record {
 
 public:
   Record(std::ostream &s, ContextInfo input_info, std::mutex &Logging_lock)
-      : stream(s), info(input_info), lock(Logging_lock) {
+      : stream(s), info(input_info), lock(Logging_lock), hash(0) {
```

Comment:
this could be useful in `get_state` function of  LogRecord class. it could provide an automatic way of getting which line / state the log was generated from, based on the arguments that that line prints. Of course it is a bit nuanced than that, but it serves as a very good approximation for this thing i'm calling 'state'

https://github.com/yogeshg/clayer/blob/yo/include/property.h#L86